sql: panic: repeat causes overflow

```
SELECT repeat('1234567890'::string, 6978072892806141784::int) 
```

```
panic: strings: Repeat count causes overflow [recovered]
	panic: panic while executing "SELECT repeat('1234567890'::string, 6978072892806141784::int) ": strings: Repeat count causes overflow [recovered]
```

Found with RSG. RSG didn't find it earlier because `repeat` was excluded from testing due to the previous panic with repeat at #10545.

--- a/pkg/sql/logictest/testdata/logic_test/builtin_function
+++ b/pkg/sql/logictest/testdata/logic_test/builtin_function
@@ -203,6 +203,10 @@ empty
 statement error pq: repeat\(\): .* memory budget exceeded
 SELECT repeat('s', 9223372036854775807)
 
+# Regression for #19035.
+statement error pq: repeat\(\): requested length too large
+SELECT repeat('1234567890'::string, 6978072892806141784::int)
+
 query I
 SELECT ascii('x')
 ----

--- a/pkg/sql/parser/builtins.go
+++ b/pkg/sql/parser/builtins.go
@@ -650,11 +650,19 @@ var Builtins = map[string][]Builtin{
 				s := string(MustBeDString(args[0]))
 				count := int(MustBeDInt(args[1]))
 
-				if count < 0 {
+				ln := len(s) * count
+				// Use <= here instead of < to prevent a possible divide-by-zero in the next
+				// if block.
+				if count <= 0 {
 					count = 0
+				} else if ln/count != len(s) {
+					// Detect overflow and trigger an error.
+					return nil, pgerror.NewError(
+						pgerror.CodeProgramLimitExceededError, "requested length too large",
+					)
 				}
 
-				if err := evalCtx.ActiveMemAcc.Grow(evalCtx.Ctx(), int64(len(s)*count)); err != nil {
+				if err := evalCtx.ActiveMemAcc.Grow(evalCtx.Ctx(), int64(ln)); err != nil {
 					return nil, err
 				}
 				return NewDString(strings.Repeat(s, count)), nil

